### PR TITLE
Preview Images in a Modal

### PR DESCRIPTION
**Pull request by graygilmore**
Additionally this PR adds a shared modal partial for the admin and instructions on its usage to the style guide.

modal

### CODE REVIEW

File: backend/app/views/spree/admin/images/_image_row.html.erb [hunk 1]
```diff
@@ -7,9 +7,15 @@
...
-    <%= link_to image.attachment.url(:product) do %>
+    <span class="modal-image-trigger" data-toggle="modal" data-target="#modal-image-<%= image.id %>">
       <%= render 'spree/admin/shared/image', image: image, size: :mini %>
-    <% end %>
+    </span>
+    <%= render(
+      "spree/admin/shared/modal",
+      target: "modal-image-#{image.id}",
+      title: image.alt,
+      content: image_tag(image.attachment.url(:large))
```
Comment (tvdeyen) [inline backend/app/views/spree/admin/images/_image_row.html.erb]: Just my personal taste, but wouldn’t it be nice to wrap the image in `div` that centers the image?

**Comment by graygilmore** [comment]
Updated to include centering of the images. This will likely not happen on normal stores as the images we're selecting to go in this modal are larger than the modal itself but I'm happy to provide a fallback for smaller images.

**Comment by graygilmore** [comment]
@tvdeyen a separate PR makes sense to me. Unfortunately overriding the value here is really hard. I tried adding:

```scss
[id^="modal-image-"] img {
  // Images in tables automatically have a border around them but we don't want
  // that in our modal.
  border: none;
}
```

But of course it's not strong enough to override this incredible selector:

```
table tbody tr.even td img {
  border: 1px solid #e7f0fa;
}
```

**Comment by tvdeyen** [comment]
@graygilmore https://github.com/solidusio/solidus/pull/2143

**Pull request merged**